### PR TITLE
chore: deps update

### PR DESCRIPTION
**Pull request by tomassasovsky**
# Description

The package needed to have its dependencies updated, mainly `analyzer`, as it's used in many other packages and incompatibilities arise when adding these to the project dependencies.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [X] Ran `dart analyze .`
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas --> no code added, just modified the existing one to be compatible with the new APIs
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/src/commands/chat_command.dart [hunk 1]
```diff
@@ -155,7 +155,7 @@ mixin ChatGroupMixin implements IChatCommandComponent {
   String get fullName =>
       (parent == null || parent is! ICommandRegisterable
           ? ''
-          : (parent as ICommandRegisterable).name + ' ') +
+          : '${(parent as ICommandRegisterable).name} ') +
```
Comment (tomassasovsky) [inline lib/src/commands/chat_command.dart:158]: this fixes a linter warning

File: pubspec.yaml [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+version: 4.3.1
 description: A framework for easily creating slash commands and text commands for Discord using the nyxx library.
 
 homepage: https://github.com/nyxx-discord/nyxx_commands/blob/main/README.md
 repository: https://github.com/nyxx-discord/nyxx_commands
 issue_tracker: https://github.com/nyxx-discord/nyxx_commands/issues
 
 environment:
-  sdk: '>=2.17.0 <3.0.0'
+  sdk: ">=2.17.0 <3.0.0"
```
Comment (abitofevrything) [inline pubspec.yaml:10]: Tiny nit but I'd rather keep the single quotes as that is the default

**Review by abitofevrything — APPROVED** [review]

File: pubspec.yaml (hunk 2 above)
Comment (tomassasovsky) [reply pubspec.yaml:10]: for sure! I think my code editor may have changed this automatically dunno why

**Review by l7ssha — APPROVED** [review]

**Review by abitofevrything — APPROVED** [review]

**Comment by tomassasovsky** [comment]
Thanks!

**Pull request merged**